metadata-ingestion/src/datahub/ingestion/source/unity/ge_profiler.py

```diff
@@ -0,0 +1,170 @@
+        # Extra default SQLAlchemy option for better connection pooling and threading.
+        # https://docs.sqlalchemy.org/en/14/core/pooling.html#sqlalchemy.pool.QueuePool.params.max_overflow
+        self.config.options.setdefault(
+            "max_overflow", self.profiling_config.max_workers
+        )
+
+        url = self.config.get_sql_alchemy_url()
+        engine = create_engine(url, **self.config.options)
+        conn = engine.connect()
+
+        profile_requests = []
+        with ThreadPoolExecutor(
+            max_workers=self.profiling_config.max_workers
+        ) as executor:
+            futures = [
+                executor.submit(
+                    self.get_unity_profile_request,
+                    UnityCatalogSQLGenericTable(table),
+                    conn,
+                )
+                for table in tables
+            ]
+
+            try:
+                for i, completed in enumerate(
+                    as_completed(futures, timeout=self.profiling_config.max_wait_secs)
+                ):
+                    profile_request = completed.result()
+                    if profile_request is not None:
+                        profile_requests.append(profile_request)
+                    if i > 0 and i % 100 == 0:
+                        logger.info(f"Finished table-level profiling for {i} tables")
+            except TimeoutError:
+                logger.warning("Timed out waiting to complete table-level profiling.")
+
+        if len(profile_requests) == 0:
+            return
+
+        yield from self.generate_profile_workunits(
+            profile_requests,
+            max_workers=self.config.profiling.max_workers,
+            platform=self.platform,
+            profiler_args=self.get_profile_args(),
+        )
+
+    def get_dataset_name(self, table_name: str, schema_name: str, db_name: str) -> str:
+        # Note: unused... ideally should share logic with TableReference
+        return f"{db_name}.{schema_name}.{table_name}"
+
+    def get_unity_profile_request(
+        self, table: UnityCatalogSQLGenericTable, conn: Connection
+    ) -> Optional[TableProfilerRequest]:
+        # TODO: Reduce code duplication with get_profile_request
+        skip_profiling = False
+        profile_table_level_only = self.profiling_config.profile_table_level_only
+
+        dataset_name = table.ref.qualified_table_name
+        try:
+            table.size_in_bytes = _get_dataset_size_in_bytes(table, conn)
+        except Exception as e:
+            logger.warning(f"Failed to get table size for {dataset_name}: {e}")
+
+        if table.size_in_bytes is None:
+            self.report.num_profile_missing_size_in_bytes += 1
+        if not self.is_dataset_eligible_for_profiling(
+            dataset_name,
+            size_in_bytes=table.size_in_bytes,
+            last_altered=table.last_altered,
+            rows_count=0,  # Can't get row count ahead of time
```

Comment:
shouldn't we pass None here?

```suggestion
                rows_count=None,
```

Reply:
Yeah, looks like we can do this now. We don't for some reason in `sql_generic_profiler.py`